build-tools/build.sh

```diff
@@ -16,74 +16,106 @@ buildToolsFolder="$(dirname "$0")"
-
-cd "$OLDPWD"
-
-# Generate the tar archive
-for stackDir in $registryFolder/stacks/*/
-do
-  cd $stackDir
-  # Find the files to add to the tar archive
-  tarFiles=$(find . \( -not -name 'devfile.yaml' \
-    -a -not -name "meta.yaml" \
-    -a -not -name "*.vsx" \
-    -a -not -name "." \
-    -a -not -name "logo.svg" \
-    -a -not -name "logo.png" \) -maxdepth 1)
-
-  # There are files that need to be pulled into a tar archive
-  if [[ ! -z $tarFiles ]]; then
-    tar -czvf archive.tar $tarFiles
-    rm -rf $tarFiles
+  # Build the index generator/validator
+  echo "Building index-generator tool"
+  ./build.sh
+  if [ $? -ne 0 ]; then
+    echo "Failed to build index-generator tool"
+    return 1
   fi
+  echo "Successfully built the index-generator tool\n"
+
   cd "$OLDPWD"
-done
 
-# Run the index generator tool
-echo "Generate the devfile registry index"
-$generatorFolder/index-generator $registryFolder/stacks $registryFolder/index.json
-if [ $? -ne 0 ]; then
-  echo "Failed to build the devfile registry index"
-  cleanup_and_exit 1
+  # Generate the tar archive
+  for stackDir in $outputFolder/stacks/*/
+  do
+    cd $stackDir
+    # Find the files to add to the tar archive
+    tarFiles=$(find . \( -not -name 'devfile.yaml' \
+      -a -not -name "meta.yaml" \
+      -a -not -name "*.vsx" \
+      -a -not -name "." \
+      -a -not -name "logo.svg" \
+      -a -not -name "logo.png" \) -maxdepth 1)
+
+    # There are files that need to be pulled into a tar archive
+    if [[ ! -z $tarFiles ]]; then
+      tar -czvf archive.tar $tarFiles
+      rm -rf $tarFiles
+    fi
+    cd "$OLDPWD"
+  done
+
+  # Run the index generator tool
+  echo "Generating the devfile registry index"
+  $generatorFolder/index-generator $outputFolder/stacks $outputFolder/index.json
+  if [ $? -ne 0 ]; then
+    echo "Failed to build the devfile registry index"
+    return 1
+  fi
+  echo "Successfully built the devfile registry index\n"
+}
+
+# check_params validates that the arguments passed into the script are valid
+# The first parameter must point to a valid devfile registry folder, containing a stacks folder
+# The second parameter must point to an empty output folder, or a folder that does not yet exist.
+check_params() {
+  # If the output folder does not have a stacks folder, we cannot do the build, so exit out
+  if [ ! -d "$registryRepository/stacks" ]; then
+    echo "A valid devfile registry was not passed in. Please specify a devfile registry folder scontaining a stacks folder."
+    display_usage
+    exit 1
+  fi
+
+  # If the output registry folder does not exist, create it.
+  if [ ! -d $outputFolder ]; then
+    mkdir -p $outputFolder
+  fi
+
+  # If the speicifed output folder is not empty, exit.
+  if [ ! -z "$(ls -A $outputFolder)" ]; then
+    echo "The specified destination folder is not empty. Please specify an empty folder."
+    display_usage
+    exit 1
+  fi
+}
+
+# Check if a registry repository folder was passed in, if not, exit
```

Comment:
Nit: Should the comment be `Check if a registry repository folder and a output folder were passed in, if not, exit`?

Reply:
Yeah, true. I'll update it.